### PR TITLE
feat: Map upload validator

### PR DESCRIPTION
**Pull request by alexlambson**
- Map upload endpoint
- Validate map files
- Add cncnet map ID
- extract previews

### CODE REVIEW

**Review by Metadorius — COMMENTED** [review]
Goes a bit out of scope of my knowledge, but for me looks good. I haven't looked in details at the model though. Would it be possible to create a ER diagram for easy reviewing?

File: kirovy/utils/service_utils.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+from kirovy import typing as t, exceptions as e
+from kirovy.constants import GameEngines
+from kirovy.models import CncGame
+from kirovy.services.cnc_gen_2_services import CncGen2MapParser
+
+
+def get_map_parser_for_game(game: CncGame) -> t.Type[CncGen2MapParser]:
+    """Get the map parser for a game.
+
+    Right now it basically just returns the gen 2 map parser, but we'll need this when we add SAGE support.
```
[the remark this answers is not in the document]
Comment (alexlambson) [reply kirovy/utils/service_utils.py:10]: Oh does Gen 1 work differently? Good to know

File: kirovy/constants/__init__.py [hunk 2]
```diff
@@ -1,6 +1,138 @@
...
+        return user_group == cls.KANE
+
+
+class GameSlugs(enum.StrEnum):
+    """The slugs for each game / total conversion mod.
+
+    These **must** be unique. They are in constants because we need them to determine which parser to use
+    for maps and collections.
+
+    ## IF YOU CHANGE THESE THEN YOU **MUST** MAKE A DATA MIGRATION!!!
```
[the remark this answers is not in the document]
Comment (alexlambson) [reply kirovy/constants/__init__.py:94]: Good point. I'll add that

File: kirovy/constants/__init__.py [hunk 3]
```diff
@@ -1,6 +1,138 @@
...
+
+
+class CncnetUserGroup:
+    """A class with the user roles in the CnCNet Ladder API, and helpers for roles."""
+
+    USER: t.Literal["User"] = "User"
+    MOD: t.Literal["Moderator"] = "Moderator"
+    ADMIN: t.Literal["Admin"] = "Admin"
+    GOD: t.Literal["God"] = "God"
+    KANE: t.Literal["God"] = GOD
```
[the remark this answers is not in the document]
Comment (alexlambson) [reply kirovy/constants/__init__.py:19]: Kane dealt with him

**Comment by alexlambson** [comment]
That's a good idea. I'll make one after work. I think pycharm has a tool for it

**Comment by alexlambson** [comment]
@Metadorius, sorry forgot to upload. 
kirovy-diagram

**Pull request merged**